src/data/repositories/consts/DiseaseOutbreakConstants.ts

```diff
@@ -56,21 +60,45 @@ export function isStringInDiseaseOutbreakCodes(code: string): code is KeyCode {
     return (Object.values(diseaseOutbreakCodes) as string[]).includes(code);
 }
 
-export function getHazardTypeValue(hazardType: string): HazardType {
-    const hazardTypeString = Object.keys(hazardTypeCodeMap).find(
-        key => hazardTypeCodeMap[key as HazardType] === hazardType
-    );
-    return hazardTypeString ? (hazardTypeString as HazardType) : "Unknown";
+export function mapValueToDataSource(value: string): DataSource | undefined {
+    switch (value) {
+        case "IBS":
+            return DataSource.IBS;
+        case "EBS":
+            return DataSource.EBS;
+        default:
+            return undefined;
+    }
+}
+
+export function mapValueToIncidentStatus(value: string): IncidentStatus | undefined {
+    switch (value) {
+        case "WATCH":
+            return IncidentStatus.WATCH;
+        case "ALERT":
+            return IncidentStatus.ALERT;
+        case "RESPOND":
+            return IncidentStatus.RESPOND;
+        case "CLOSED":
+            return IncidentStatus.CLOSED;
+        case "DISCARDED":
+            return IncidentStatus.DISCARDED;
+        default:
+            return undefined;
+    }
```

Comment:
Not wrong or anything, just as a note, a `Record<string, IncidentStatus>` can also be used for direct mappings.

Reply:
totally agree!